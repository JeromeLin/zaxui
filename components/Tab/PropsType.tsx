interface BaseProps {
  theme?: 'default' | 'primary' | 'info' | 'success' | 'warning' | 'error';
  lineWidth?: string | number;
  disabled?: boolean;
  canSwipe?: boolean;
  onChange?: (index: number) => void;
  children?: any;
}

export interface TabProps extends BaseProps {
  prefixCls?: string;
  className?: string;
}

export interface TabPannelProps {
  prefixCls?: string;
  className?: string;
}
