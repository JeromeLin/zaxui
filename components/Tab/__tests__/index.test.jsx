import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tab from '../index';

describe('Tab', () => {
  it('renders correctly', () => {
    const onChange = jest.fn();

    const wrapper = render(
      <Tab canSwipe onChange={onChange}>
        <Tab.Panel title="选项卡1">
          <div>试试点我左滑</div>
        </Tab.Panel>
        <Tab.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tab.Panel>
      </Tab>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
