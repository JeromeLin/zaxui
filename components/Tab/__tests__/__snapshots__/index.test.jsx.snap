// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders correctly 1`] = `
<div
  class="za-tab theme-primary"
>
  <div
    class="za-tab-header"
  >
    <ul
      role="tablist"
    >
      <li
        class="za-tab-header-item active"
        role="tab"
      >
        选项卡1
      </li>
      <li
        class="za-tab-header-item"
        role="tab"
      >
        选项卡2
      </li>
    </ul>
    <div
      class="za-tab-line"
      style="width:50%;left:0%;"
    />
  </div>
  <div
    class="za-tab-container"
  >
    <div
      class="za-swipe"
    >
      <div
        class="za-swipe-items"
        style="white-space:nowrap;"
      >
        <div
          class="za-swipe-item"
        >
          <div>
            试试点我左滑
          </div>
        </div>
        <div
          class="za-swipe-item"
        >
          <div>
            试试点我右滑
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
